Source/MQTTnet/Implementations/MqttTcpChannel.cs

```diff
@@ -150,7 +156,7 @@ public async Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancellat
                 }
 
                 // Workaround for: https://github.com/dotnet/corefx/issues/24430
-                using (cancellationToken.Register(Dispose))
```

Comment:
In the above code I agree that the lambda is not required but here we already have a method without wrapping it. Do we really need that optimization here as well?

Reply:
From my profiling it showed up in `WriteAsync` (L187), so it's just duplicated here.

But when the conditional compilation is added for .NET Core 3.0 and newer, this won't be hit. So for the legacy code we could
* keep the state of _master_-branch
* keep the cached delegate of PR's current state

I'm leaning to the latter, as the change isn't huge and so older targets can benefit from that change too.

Reply:
Ah, now I see your concern. Yes we need it. Whether it's a delegate created from a method group or a lambda, the delegate won't be cached. Checked the IL for both.

Reply:
OK then we should keep it.